### PR TITLE
Javascript -III challenge

### PR DESCRIPTION
**Pull request by ogwurujohnson**

### CODE REVIEW

File: assignments/prototypes.js [hunk 1]
```diff
@@ -32,79 +53,129 @@
...
+  --this.healthPoints;
+  if (this.healthPoints <= 0) {
+    console.log(`Remaining points- ${this.healthPoints}`);
+    return `${this.destroy()}
+      --------------------------------`;
+  } else {
+    console.log(`Remaining points- ${this.healthPoints}`);
+    return `${this.takeDamage()}
+        -----------------------------`;
+  }
```
Comment (mattshardman) [inline assignments/prototypes.js]: As these two removeHealthPoints prototypes are identical, you could separate them into its own function and assign that to the prototype of both hero and villain, then just pass the person into the function when it is called.

Something like this:

```Hero.prototype = { ...Humanoid.prototype, removeHealthPoints };```

This helps keep your code dry.

**Comment by mattshardman** [comment]
Awesome job Johnson!

* You have correctly created the required objects using constructor functions
* Your constructors are correctly inherit from each other
* Your git workflow is clean and fluid
* You code is well organised and easy to read
* You have tackled the stretch challenge and created a villain
* Good answers to questions

Overall great job you clearing have a good understanding of constructor functions and prototypal inheritance. Nice fight with Melvine too lol

File: assignments/prototypes.js (hunk 1 above)
Comment (ogwurujohnson) [reply assignments/prototypes.js]: Yeah,  you are right.  Have taken note of that,  would be mindful of such from now on

**Comment by ogwurujohnson** [comment]
Thanks Matt,  had to make sure Melvine was defeated as a villain

**Pull request merged**